Libraries/PeriphDrivers/Source/SMON/smon_reva.h

```diff
@@ -30,145 +30,61 @@
-    SMON_REVA_EXTSENSOR_3 = MXC_F_SMON_REVA_EXTSCN_EXTS_EN3,
-    SMON_REVA_EXTSENSOR_4 = MXC_F_SMON_REVA_EXTSCN_EXTS_EN4,
-    SMON_REVA_EXTSENSOR_5 = MXC_F_SMON_REVA_EXTSCN_EXTS_EN5,
-} mxc_smon_reva_ext_sensor_t;
-
-/**
- * @brief   Enum for Clock Divider
- *
- */
-typedef enum {
-    SMON_REVA_CLK_DIVIDE_1 = MXC_S_SMON_REVA_EXTSCN_DIVCLK_DIV1,
-    SMON_REVA_CLK_DIVIDE_2 = MXC_S_SMON_REVA_EXTSCN_DIVCLK_DIV2,
-    SMON_REVA_CLK_DIVIDE_4 = MXC_S_SMON_REVA_EXTSCN_DIVCLK_DIV4,
-    SMON_REVA_CLK_DIVIDE_8 = MXC_S_SMON_REVA_EXTSCN_DIVCLK_DIV8,
-    SMON_REVA_CLK_DIVIDE_16 = MXC_S_SMON_REVA_EXTSCN_DIVCLK_DIV16,
-    SMON_REVA_CLK_DIVIDE_32 = MXC_S_SMON_REVA_EXTSCN_DIVCLK_DIV32,
-    SMON_REVA_CLK_DIVIDE_64 = MXC_S_SMON_REVA_EXTSCN_DIVCLK_DIV64,
-} mxc_smon_reva_clk_divide_t;
-
-/**
- * @brief   Enum for Frequency Divider
- *
- */
-typedef enum {
-    SMON_REVA_FREQ_DIVIDE_4 = MXC_S_SMON_REVA_EXTSCN_EXTFRQ_FREQ2000HZ,
-    SMON_REVA_FREQ_DIVIDE_8 = MXC_S_SMON_REVA_EXTSCN_EXTFRQ_FREQ1000HZ,
-    SMON_REVA_FREQ_DIVIDE_16 = MXC_S_SMON_REVA_EXTSCN_EXTFRQ_FREQ500HZ,
-    SMON_REVA_FREQ_DIVIDE_32 = MXC_S_SMON_REVA_EXTSCN_EXTFRQ_FREQ250HZ,
-    SMON_REVA_FREQ_DIVIDE_64 = MXC_S_SMON_REVA_EXTSCN_EXTFRQ_FREQ125HZ,
-    SMON_REVA_FREQ_DIVIDE_128 = MXC_S_SMON_REVA_EXTSCN_EXTFRQ_FREQ63HZ,
-    SMON_REVA_FREQ_DIVIDE_256 = MXC_S_SMON_REVA_EXTSCN_EXTFRQ_FREQ31HZ,
-} mxc_smon_reva_freq_divide_t;
-
-/**
- * @brief   Voltage Monitor Thresholds
- *
- */
-typedef enum {
-    SMON_REVA_VTM_THRESHOLD_1_6, ///< 1.6 V
-    SMON_REVA_VTM_THRESHOLD_2_2, ///< 2.2 V
-    SMON_REVA_VTM_THRESHOLD_2_8, ///< 2.8 V
-} mxc_smon_reva_vtm_t;
-
-/**
- * @brief   Temperature Sensor Thresholds
- *
- */
-typedef enum {
-    SMON_REVA_TEMP_THRESHOLD_NEG_50, ///< -50 *C
-    SMON_REVA_TEMP_THRESHOLD_NEG_30, ///< -30 *C
-} mxc_smon_reva_temp_t;
-
-/**
- * @brief   Digital Fault Interrupt mode
- *
- */
-typedef enum {
-    SMON_REVA_DFD_INTERRUPT_NMI, ///< DRS/NMI
-    SMON_REVA_DFD_INTERRUPT_PFW, ///< PFW IRQ
-} mxc_smon_reva_interrupt_mode_t;
-
-/**
- * @brief   Digital Fault Low Power mode
- *
- */
-typedef enum {
-    SMON_REVA_DFD_LOWPOWER_ENABLE, ///< DFD enabled during LowPower mode
-    SMON_REVA_DFD_LOWPOWER_DISABLE, ///< DFD disabled during LowPower mode
-} mxc_smon_reva_lowpower_mode_t;
-
-/**
- * @brief   Register to check if busy
- *
- */
-typedef enum {
-    SMON_REVA_EXTSENSOR = MXC_F_SMON_REVA_SECST_EXTSRS,
-    SMON_REVA_INTSENSOR = MXC_F_SMON_REVA_SECST_INTSRS,
-    SMON_REVA_SECALARM = MXC_F_SMON_REVA_SECST_SECALRS,
-} mxc_smon_reva_busy_t;
-
-/**
- * @brief   The information required to configure an external sensor
- *
- */
-typedef struct {
-    mxc_smon_reva_ext_sensor_t sensorNumber;
-    mxc_smon_reva_clk_divide_t clockDivide;
-    mxc_smon_reva_freq_divide_t freqDivide;
-    uint8_t errorCount;
-    uint8_t data;
-} mxc_smon_reva_ext_cfg_t;
-
-int MXC_SMON_RevA_ExtSensorEnable(mxc_smon_reva_regs_t *smon, mxc_smon_reva_ext_cfg_t *cfg,
+#if TARGET_NUM == 32572
```

Comment:
Is the #if necessary?  Can't smon_meXX.c files handle whether to call these functions or just return E_NOT_SUPPORTED?

Reply:
Nope, you're right. Originally, the new structs in the ME55 weren't added in the other parts yet which caused build errors for the ES17 and ME13A.